lib/bundler/cli/install.rb

```diff
@@ -192,7 +192,9 @@ def normalize_groups
     def normalize_settings
       Bundler.settings[:path]                = nil if options[:system]
       Bundler.settings[:path]                = "vendor/bundle" if options[:deployment]
-      Bundler.settings[:path]                = options["path"] if options["path"]
+
+      Bundler.settings[:path]                = Bundler.feature_flag.path_relative_to_cwd? ? File.expand_path(options["path"]) : options["path"] if options["path"]
```

Comment:
ping on this -- I think you'll want to expand this where its used, rather than in the setting